### PR TITLE
refactor: Use iceberg-rust to replace icelake

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Part of https://github.com/datafuselabs/databend/issues/15809

The real migrate work happen here.

This PR adds two catalog support for iceberg:

- rest: The REST catalog support
- hms: The Hive Metastore support

To use those catalogs, user will need to create the catalog by:

```sql
CREATE CATALOG iceberg_ctl
TYPE=ICEBERG                             -- use ICEBERG as catalog type
CONNECTION=(
    TYPE='rest'                          -- the type of iceberg catalog.
    ADDRESS='http://127.0.0.1:8181'      -- the address of rest catalog.
    WAREHOUSE='s3://icebergdata/demo'    -- the warehouse we want to connect.
);
```

Please note, iceberg has multiple catalog support by itself.

Currently, Iceberg requires a catalog to connect, so I have temporarily disabled support for the `iceberg` table.

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15951)

### CODE REVIEW

**Review by drmingdrmer — COMMENTED** [review]
Are there meta data changes? and need to add a new version?
https://github.com/datafuselabs/databend/blob/4dfb07070690dc958cba30882cbac679ca0e6eff/src/meta/proto-conv/src/util.rs#L127-L132

> Reviewed 4 of 15 files at r1, 8 of 27 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15951#-:-O0y2feP96Y3GVqejT6L:bv8jhlr)* status: 9 of 30 files reviewed, all discussions resolved (waiting on @BohuTANG and @youngsofun)

**Comment by Xuanwo** [comment]
> Are there meta data changes? and need to add a new version?

Sorry for missing this part. I will add a new version.

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 2 of 15 files at r1, 10 of 27 files at r2, 2 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15951)* status: 22 of 35 files reviewed, all discussions resolved (waiting on @BohuTANG and @youngsofun)

File: tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh [hunk 1]
```diff
@@ -10,16 +10,14 @@ cat <<EOF | $BENDSQL_CLIENT_CONNECT
 CREATE CATALOG iceberg_ctl
 TYPE=ICEBERG
 CONNECTION=(
-    URL='s3://testbucket/iceberg_ctl/'
-    access_key_id ='minioadmin'
-    secret_access_key ='minioadmin'
-    ENDPOINT_URL='${STORAGE_S3_ENDPOINT_URL}'
+    TYPE='rest'
```
Comment (youngsofun) [inline tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh:13]: hms not tested

Comment (youngsofun) [inline tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh:13]: is hms ready for test?

add basic read tests?

Comment (Xuanwo) [reply tests/suites/3_stateful_iceberg/00_rest/00_0000_create_and_show.sh:13]: > is hms ready for test?

I plan to implement in following PRs.


> add basic read tests?

I'm still figuring out how to setup a demo table, maybe need spark. I will finish in next PR.

**Review by youngsofun — APPROVED** [review]

**Pull request merged**